### ISSUE TITLE
Consider a tls-api-passthrough

### ISSUE THREAD

**Issue by algesten** [stepancheg/rust-tls-api#31]
I recently started using tls-api. In my use case I to be able to chose between TLS implementations, but also no TLS when appropriate.

My API is:

```rust
pub async fn connect<Tls: TlsConnector>(uri: &http::Uri) -> Result<Connection, Error> { … }
```

For connecting plain `http` (not that one should), instead of having a duplicate `connect()` function with code duplication, the `TlsConnector` could be a pass through implementation.

I written one in my own repo, but would you consider a PR with it (fixed up) to sit alongside your other implementations?

https://github.com/algesten/ureq/blob/future/src/tls_pass.rs

**Comment by stepancheg** [comment]
This would be kind of misleading — if not-TLS connector would return "TlsStream" object.

That said

Implement something like "tls-api-not-tls" crate (similar to what you have shown) might be useful in certain use cases.

An alternative is to implement something like "Connector" trait (without Tls- prefix), and implement it for all of Tls connectors.

Additionally, in your crate, you may want to implement your own Connector trait with plain implementation and implementation delegating to tls-api.

TL;DR I don't object PR for "tls-api-not-tls", but I I'm not sure this is the right thing to do.

**Comment by stepancheg** [comment]
Implemented.
